File: 2256-minimum-average-difference/2256-minimum-average-difference.py
class Solution:
    def minimumAverageDifference(self, nums: List[int]) -> int:
        l = len(nums)
        sumt = [0 for i in range(l)]
        for i in range(l-2, -1, -1):
            sumt[i] += nums[i+1] + sumt[i+1]
        
        curr, ans, mini = 0, float('inf'), -1
        for i in range(l):
            curr+=nums[i]
            curr_ans = abs((curr//(i+1)) - (sumt[i]//(l-i-1))) if l-i-1>0 else abs((curr//(i+1)))
            if curr_ans < ans:
                mini = i
                ans = curr_ans
        
        return mini

File: 2256-minimum-average-difference/2256-minimum-average-difference.java
class Solution {
    public int minimumAverageDifference(int[] nums) {
        long sumt=0;
        for(int n: nums) sumt+=n;
        long mini = 1000000000, curr=0;
        int minind=-1,l=nums.length;
        
        for(int i=0; i<l; ++i){
            curr+=nums[i];
            sumt-=nums[i];
            long currDiff = (i!=l-1) ? Math.abs(curr/(i+1) - sumt/(l-i-1)) : Math.abs(curr/(i+1)) ;
            if(currDiff<mini){
                mini=currDiff;
                minind=i;
            }
        }
        return minind;
    }
}